weblate/api/views.py

```diff
@@ -646,6 +657,55 @@ def file(self, request, **kwargs):
 
         return Response(data={"result": True})
 
+    @action(
+        detail=True, methods=["post"],
+    )
+    def units(self, request, **kwargs):
+        obj = self.get_object()
+        if "unit_pk" not in request.data:
+            raise ParseError("Missing unit_pk parameter")
+
+        try:
+            source_string = obj.component.source_translation.unit_set.get(
+                pk=int(request.data["unit_pk"])
+            )
+        except (Unit.DoesNotExist, ValueError) as error:
+            return Response(
+                data={"result": "Unsuccessful", "detail": force_str(error)},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        obj.units.add(source_string)
+        serializer = ScreenshotSerializer(obj, context={"request": request})
+
+        return Response(serializer.data, status=status.HTTP_200_OK,)
+
+    def create(self, request, *args, **kwargs):
+        if not request.user.has_perm("screenshot.add"):
```

Comment:
```suggestion
        if not request.user.has_perm("screenshot.add", component):
```
This should be scoped per component, it will also need to move some other code above.